pkg/gameserverallocations/controller.go

```diff
@@ -51,8 +51,8 @@ type Controller struct {
 	allocator  *Allocator
 }
 
-// NewController returns a controller for a GameServerAllocation
-func NewController(apiServer *apiserver.APIServer,
+// NewExtensions returns a controller for a GameServerAllocation
+func NewExtensions(apiServer *apiserver.APIServer,
```

Comment:
Why is this being renamed? It is no longer a controller? The return type is still `*Controller`.

Reply:
I asked for the rename because it follows the pattern of the other "NewExtensions" and makes it clear which side of the divide. (I missed the return type from review fatigue.)

Reply:
Hmm, I see how it can be confusing without changing the struct name. We could either change the struct as well but not sure if that would cause any confusion on other parts or change it back?

Reply:
🤔 I'd probably be inclined to move the whole type to being *Extension, since that would align with the responsibilities of all the other Extension types.